contrib/babelfishpg_tsql/antlr/TSqlParser.g4

```
@@ -3956,16 +3955,34 @@ geospatial_func_arg
     | STINTERSECTS
     ;
 
+geospatial_static_method
+    : STGEOMFROMTEXT
+    | STPOINTFROMTEXT
+    | POINT
+    ;
+
 hierarchyid_methods
     : method=( GETANCESTOR | GETDESCENDANT | GETLEVEL | ISDESCENDANTOF | READ | GETREPARENTEDVALUE | TOSTRING ) LR_BRACKET expression_list? RR_BRACKET
     ;
 
-hierarchyid_coloncolon_methods
-    : id colon_colon  method=(GETROOT | PARSE) LR_BRACKET expression? RR_BRACKET
+hierarchyid_static_method
+    : GETROOT
+    | PARSE
+    ;
+
+datatype_static_method
+    : hierarchyid_static_method
+    | geospatial_static_method
+    ;
+
+datatype_field_or_method
+    : method=id (LR_BRACKET expression_list? RR_BRACKET)
+    | field=id
     ;
 
-spatial_coloncolon_methods
-    : data_type colon_colon function_call
+datatype_coloncolon_methods
+    : data_type colon_colon datatype_static_method LR_BRACKET expression_list? RR_BRACKET
+    | data_type colon_colon datatype_field_or_method
```

Shouldn't we keep this as `data_type colon_colon function_call`?

---

No we cannot keep this as `data_type colon_colon function_call`, since `function_call` contains following syntax
```
function_call
    : ranking_windowed_function                         
    | aggregate_windowed_function                      
    | analytic_windowed_function                       
    | spatial_proc_name_server_database_schema LR_BRACKET function_arg_list? RR_BRACKET 
    | xml_proc_name_table_column LR_BRACKET expression_list? RR_BRACKET
    | func_proc_name_server_database_schema LR_BRACKET allOrDistinct=(DISTINCT|ALL)? function_arg_list? RR_BRACKET 
    | built_in_functions                               
    | freetext_function                                
    | NEXT VALUE FOR full_object_name		         
    | odbc_scalar_function           
    | partition_function_call                           
    ;
```
and not all of them are valid syntax after colon_colon in a `data_type colon_colon method` syntax.

---

So, we can throw the appropriate error, as we are doing currently